reftests/src/tests/attributes/_2_ledger.rs

```diff
@@ -1,16 +1,547 @@
+        Ok(total.into())
+    }
+
+    pub async fn events_list(&self) -> Result<u128, String> {
+        let payload = "{}".to_string();
+        let envelope = anonymous_message("events.list", payload);
+        let response = send(&self.test_config, envelope).await;
+
+        let payload = response.payload.expect("No payload from status");
+        let response: BTreeMap<u8, Value> =
+            ciborium::de::from_reader(payload.as_slice()).expect("Invalid payload.");
+
+        let response_payload = response
+            .get(&4)
+            .expect("Response return value was not found")
+            .as_bytes()
+            .ok_or_else(|| {
+                format!(
+                    "[Response: {:?} -> was expected to be Bytes]",
+                    response.get(&4).unwrap()
+                )
+            })?;
+
+        let events_list_returns: ListReturns =
+            minicbor::decode(response_payload.as_slice()).unwrap();
+
+        let total = events_list_returns.nb_events;
+        Ok(total.into())
+    }
+}
+
+#[test_case]
+async fn send_to_bad_identity(test_config: TestConfig) -> TestCaseResult {
+    // If the config file is not present, it should skip the test.
+    // This should return something that skips
+    let ledger_config = test_config.read_config("ledger".to_string());
+
+    // This should probably make a call to `status` endpoint of the server,
+    // but should definitely cache the result (avoid calling it everytime).
+    let client = LedgerClient::new(test_config, ledger_config).await;
+
+    async fn send_test(client: LedgerClient) -> Result<(), String> {
+        let akward_identity = "some_akward_impossible_identity".to_string();
+
+        // Test balance query
+        // Balance should verify that the server has the attribute `2`.
+        if !has_attribute(2, &client.test_config).await {
+            return Err("Server does not support ledger attribute.".to_string());
+        }
+
+        // Checking faucet has enough funds for following tests.
+        assert!(
+            client
+                .balance(
+                    client.ledger_config.faucet.to_string(),
+                    client.ledger_config.symbol
+                )
+                .await
+                .unwrap()
+                > 10000
+        );
+
+        // Test sending command
+        // Balance should verify that the server has the attribute `6`.
+        if !has_attribute(6, &client.test_config).await {
+            return Err("Server does not support ledger commands attribute.".to_string());
+        }
+
+        // Sending 10000 from faucet to odd identity
+        match client
+            .send(
+                None,
+                client.ledger_config.faucet.to_string(),
+                akward_identity.to_owned(),
+                10000,
+                client.ledger_config.symbol,
+                Some(client.ledger_config.faucet_pk.to_owned()),
+            )
+            .await
```

Comment:
You can simplify this using `unwrap_err()` and `is_err()`.

See tests in many-framework